.jenkins/infrastructure/docker/build_windows_docker_images.Jenkinsfile

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+OECI_LIB_VERSION = env.OECI_LIB_VERSION ?: "master"
+oe = library("OpenEnclaveCommon@${OECI_LIB_VERSION}").jenkins.common.Openenclave.new()
+
+AGENTS_LABEL = params.AGENTS_LABEL
+TIMEOUT_MINUTES = params.TIMEOUT_MINUTES ?: 240
+
+INTERNAL_REPO = params.INTERNAL_REPO ?: "https://oejenkinscidockerregistry.azurecr.io"
+INTERNAL_REPO_CREDS = params.INTERNAL_REPO_CREDS ?: "oejenkinscidockerregistry"
+DOCKERHUB_REPO_CREDS = params.DOCKERHUB_REPO_CREDS ?: "oeciteamdockerhub"
+
+def buildWindowsDockerContainers() {
+    node(AGENTS_LABEL) {
+        timeout(TIMEOUT_MINUTES) {
+            stage("Checkout") {
+                cleanWs()
+                checkout scm
+            }
+            stage("Build Windows Server 2019 Docker Image") {
+                oe2019 = oe.dockerImage("oetools-ws2019:${DOCKER_TAG}", ".jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile")
+                puboe2019 = oe.dockerImage("oeciteam/oetools-ws2019:${DOCKER_TAG}", ".jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile")
+            }
+            stage("Push to OE Docker Registry") {
+                docker.withRegistry(INTERNAL_REPO, INTERNAL_REPO_CREDS) {
+                    oe.exec_with_retry { oe2019.push() }
+                    if(TAG_LATEST == "true") {
+                        oe.exec_with_retry { oe2019.push('latest') }
+                    }
+                }
+            }
+            stage("Push to OE Docker Hub Registry") {
+                docker.withRegistry('', DOCKERHUB_REPO_CREDS) {
+                    if(TAG_LATEST == "true") {
+                        oe.exec_with_retry { puboe2019.push() }
```

Comment:
I think there is a bracket issue here.

Line 36 should be on line 35, and 35 on 36.

Reply:
I think this is the intended behaviour we want; based on the Linux container build it seems like we only want to update the Docker Hub registry if `TAG_LATEST` is true. We do update the internal docker repository in line 27, with or without that flag.

Reply:
It may be intended but is it best or is it done this way for a legacy reasons? 

In a year we may not be the ones maintaining this and having debuggable containers that are pullable outside of Azure infrastructure I believe to be preferable to simplify debugging. I'll leave the decision to you, but I would prefer to see pushing tagged containers regardless of latest every time and then latest dependent on the conditional.

Reply:
I'll give the tags a more thorough look, and I'll address it in a future PR.